test/unit-test/core_sntp_serializer_utest.c

```diff
@@ -474,17 +474,23 @@ void test_DeserializeResponse_AcceptedResponse_ClockOffset_Edge_Cases( void )
      * is ahead of the client in this special case, and thus return the maximum
      * signed 32 bit integer as the clock-offset.
      */
-    serverTime.seconds = clientTime.seconds - INT32_MAX - 1;
+    /* Case when "Server Time - Client Time" results in negative value. */
+    serverTime.seconds = clientTime.seconds + INT32_MAX + 1;
     testClockOffsetCalculation( &clientTime, &serverTime,
                                 &serverTime, &clientTime,
                                 SntpSuccess,
                                 INT32_MAX );
-    serverTime.seconds = clientTime.seconds + INT32_MAX + 1;
+    /* Test when "Server Time - Client Time" results in positive value. */
+    serverTime.seconds = UINT32_MAX;
```

Comment:
These resets indicate that probably we are writing too much tests in one test. This test should be broken in multiple tests: 

1. The client is 68 years behind of server time .
2. The client is 68 years ahead of server time .
3. When "Server Time - Client Time" results in negative value.
4. When "Server Time - Client Time" results in positive value.
5. Server time is INT32_MAX seconds ahead.
6. Server time is INT32_MAX seconds behind.
7. The send timestamps overflow but receive timestamps do not overflow.
8. The send timestamps do not overflow but receive timestamps overflow.

Reply:
I can look to commonize the resets in a utility function/macro to avoid the code repetition. Would it be fine to keep that for a separate hygiene PR?